### PR TITLE
style: Move -moz-force-broken-image-icon out of mako

### PR DESCRIPTION
**Pull request by canova**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix part of #19015.

- [X] These changes do not require tests because these are just refactoring.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19279)

### CODE REVIEW

**Comment by canaltinova** [comment]
r? @emilio or @Manishearth 
That should be easy to review :)

File: components/style/values/specified/ui.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+use parser::{Parse, ParserContext};
+use std::fmt;
+use style_traits::{ParseError, StyleParseErrorKind, ToCss};
+
+/// Specified value of `-moz-force-broken-image-icon`
+#[derive(Clone, Copy, Debug, MallocSizeOf, PartialEq, ToComputedValue)]
+pub struct MozForceBrokenImageIcon(pub bool);
+
+impl MozForceBrokenImageIcon {
+    /// Return initial value of -moz-script-min-size.
```
Comment (CYBAI) [inline components/style/values/specified/ui.rs]: Typo in the comment

File: components/style/values/specified/ui.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    #[inline]
+    pub fn initial_value() -> MozForceBrokenImageIcon {
```
Comment (CYBAI) [inline components/style/values/specified/ui.rs]: Will it be better to name the `initial_value` in `false`?

Comment (emilio) [inline components/style/values/specified/ui.rs]: Yup, +1 for naming this method `false`.

File: components/style/values/specified/ui.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        MozForceBrokenImageIcon(false)
+    }
+}
+
+impl Parse for MozForceBrokenImageIcon {
+    fn parse<'i, 't>(
+        _context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<MozForceBrokenImageIcon, ParseError<'i>> {
+        match input.expect_integer()? {
```
Comment (emilio) [inline components/style/values/specified/ui.rs]: Please add a note here saying that we intentionally don't support `calc`.

File: components/style/values/specified/ui.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+impl From<u8> for MozForceBrokenImageIcon {
+    fn from(bits: u8) -> MozForceBrokenImageIcon {
+        MozForceBrokenImageIcon(bits == 1)
+    }
+}
+
+impl From<MozForceBrokenImageIcon> for u8 {
+    fn from(v: MozForceBrokenImageIcon) -> u8 {
+        match v.0 {
```
Comment (emilio) [inline components/style/values/specified/ui.rs]: What about: `if v.0 { 1 } else { 0 }`?

**Review by emilio — APPROVED** [review]
Looks good with those!

**Comment by emilio** [comment]
And glad to see you around again @canaltinova! :)

**Comment by canaltinova** [comment]
Thanks both of you for reviewing!
But we cannot name a method `false` in rust(I'm open to suggestions):
https://play.rust-lang.org/?gist=26051931f271b2b7d0803f283569717d&version=stable

Also, thanks @emilio! I will be here more often :)

**Comment by KiChjang** [comment]
@canaltinova The easiest way is often to add an underscore after the keyword, i.e. name it false_.

**Comment by canaltinova** [comment]
@KiChjang Yeah, for example, we used that solution for `box_.rs` before but personally I don't like to add underscore at the end of a function if we can use another name instead. Maybe we can use something like `false_value` to workaround that? :)

**Comment by canaltinova** [comment]
Anyway, I'm going with `false_value`, probably that's not too important to block the PR :)
@bors-servo r=emilio

**Pull request merged**